solidity.pegjs

```diff
@@ -1424,8 +1456,18 @@ SourceElements
     }
 
 SourceElement
-  = AssignmentExpression __ EOS
-  / DeclarativeExpression __ EOS
+  = exp:AssignmentExpression  __ EOS &{
+      return (
+        exp.type === 'AssignmentExpression' && 
+        exp.left.type === 'DeclarativeExpression' && 
+        exp.left.storage_location === null
+      )
+    }{

```

Comment:
Yes, it would also get rid of nasty stuff like `uint foo += 1;` getting parsed incorrectly as a state variable declaration.